api/upload.go

```diff
@@ -0,0 +1,85 @@
+package api
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/SkynetLabs/skynet-accounts/database"
+	"github.com/julienschmidt/httprouter"
+	"gitlab.com/NebulousLabs/errors"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+type (
+	// UploaderInfo gives information about a user who created an upload.
+	UploaderInfo struct {
+		UserID   primitive.ObjectID
+		Email    string
+		Sub      string
+		StripeID string
+	}
+
+	// UploadInfo gives information about a given upload.
+	UploadInfo struct {
+		Skylink    string
+		UploaderIP string
+		UploadedAt time.Time
+		UploaderInfo
+	}
+)
+
+// uploadInfoGET returns detailed information about all uploads of a given
+// skylink.
+func (api *API) uploadInfoGET(_ *database.User, w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	skylink := ps.ByName("skylink")
+	if !database.ValidSkylinkHash(skylink) {
+		api.WriteError(w, database.ErrInvalidSkylink, http.StatusBadRequest)
+		return
+	}
+
+	ctx := req.Context()
+	sl, err := api.staticDB.Skylink(ctx, skylink)
+	if err != nil {
+		api.WriteError(w, errors.AddContext(err, "failed to get skylink"), http.StatusInternalServerError)
+		return
+	}
+	// Get all uploads of this skylink.
+	ups, err := api.staticDB.UploadsBySkylinkID(ctx, sl.ID)
```

Comment:
f/u 

Definitely not a requirement for this PR but I wonder if PJ could turn this into some fancy query which fetches all uploads for a skylink id and then joins them with their users where possible. So that all of that happens with a single query.

Reply:
That's fine, I can do that as well. I'll open a F/U.